### ISSUE TITLE
你好 这个代码缺少weights_new_3.5.txt文件 怎么解决呢

### ISSUE THREAD

**Issue by Fabiabn** [wzp21142/NIDS_Reproduction#1]

**Comment by wzp21142** [comment]
issue中没有明确说明问题，以下只是对你issue的大概推测：
直接在repo中搜索weights_new_3.5.txt，似乎是想运行[SGM-CNN_UNSW-NB15.ipynb](https://github.com/wzp21142/NIDS_Reproduction/blob/f921a3404cbf71b9c638b2d548ae6d572b06823a/SGM-CNN/SGM-CNN_UNSW-NB15.ipynb)，直接看同路径下的[另一个搜索结果](https://github.com/wzp21142/NIDS_Reproduction/blob/f921a3404cbf71b9c638b2d548ae6d572b06823a/SGM-CNN/feature%20selection%EF%BC%88DAE%EF%BC%89.ipynb)可以看到这个文件的来源是DAE处理后保存的。这个算法源项目很大，所以拆成了多个jupyter notebook完成不同任务。所以你应该先运行预处理和DAE再做最后的预测任务。

**Comment by Fabiabn** [comment]
你好 我先运行了preprocess 文件 然后在运行DAE时 没有weights_new_3.5.txt文件  我已经创建了normalized 和 UNSW_NB15文件夹并导入了数据集 但是不知道是不是缺少保存文件的代码

**Comment by wzp21142** [comment]
weights_new_3.5.txt这一文件的来源是DAE训练过程中保存的模型权重
`if test_loss < min_loss:`
`	min_loss = test_loss`
`	np.savetxt('../dataset/SGM-CNN/normalized/weights_new_3.5.txt', weights, fmt = "%.6E")`
而min_loss被初始化为100，如果始终没有触发savetxt，说明整个训练过程的loss没有降到100以下过，因此没有生成文件。麻烦检查下训练过程和数据集，或者确定下是否修改了不合理的超参数

**Comment by Fabiabn** [comment]
非常感谢你的答复 但是我并没有更改你上传的代码  可能是我没看懂代码含义 可否重新上传一下您修改的py文件呢

**Comment by wzp21142** [comment]
> 非常感谢你的答复 但是我并没有更改你上传的代码 可能是我没看懂代码含义 可否重新上传一下您修改的py文件呢

本repo只做整合和补充，未对源代码做任何修改，同目录下ipynb以外的文件都和原项目一致，部分原项目（包括当前讨论的）同样在github开源，repo与目录名同名，可以自行比对
